vignettes/articles/startup_messages.Rmd

```diff
@@ -0,0 +1,149 @@
+---
+title: "How to suppress package startup messages"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{How to suppress package startup messages}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>"
+)
+```
+
+When making a reprex, you might have to load libraries that are producing some messages and warnings. The goal of this article is to suppress these, without suppressing all messages and warnings of the reprex, as they might be useful for debugging. This article aims at answering the issue [#187](https://github.com/tidyverse/reprex/issues/187). 
+
+We'll review the different possibilities to silence messages and warnings when loading a library using using a fake package that gives messages and warnings upon loading.
+
+## Creating and loading the startuppackage
```

Comment:
There is a lot of overhead associated with creating a package here in this vignette. It's a cool idea but I'm not convinced the extra work is justified. If we kept this idea, I'd have lots of comments about how we need to remove all the code about managing this package out of the vignette and into some other files.

I propose that you let go of this approach and, instead, just demonstrate startup message suppression with existing packages. The dplyr example is, in fact, the main motivator for this article, so just covering that is a tremendously helpful thing.